src/app.js

```diff
@@ -695,7 +704,7 @@
 			if (_validateNextPage(nextPage) && _canEnterNextPage(nextPage)) {
 				if (nextPage !== currentPage) {
 					gotoPage(route);
-				} else if (!!previousUrl) {
+				} else if (!!previousUrl && previousUrl() !== getCurrentUrl()) {
```

Comment:
Same here previousURL is not defined yet

Reply:
WTF have I done.

Reply:
See e3e8308227

Reply:
We need a running unit test framework... fuck Qunit...

Reply:
Yep. Would tape be good with this? Does the Framework really need to be running in a browser environement ?

Reply:
> Does the Framework really need to be running in a browser environement ?

Yes. Ideally, it would run in ALL browsers. We currently use https://karma-runner.github.io/2.0/index.html but only phantomjs is configured (it would be nice to run in REAL browsers...) Karma does not have support for tape out of the box... So I would be open to a solution with uses tape + puppeteer.